sdk/core/System.Net.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -12,6 +13,21 @@ namespace System.Net.ClientModel.Core
     /// </summary>
     public static class ModelReaderWriter
     {
+        /// <summary>
+        /// Creates an instance of a model from an instance of an anonymous type.
+        /// </summary>
+        /// <typeparam name="T">The type of the model to create.</typeparam>
+        /// <param name="jsonSerializable">An instance of an anonymous type representing the model</param>
+        /// <returns>A <typeparamref name="T"/> instance.</returns>
+        public static T? Create<T>(object jsonSerializable) where T : IJsonModel<T>
+        {
+            BinaryData json = BinaryData.FromObjectAsJson(jsonSerializable);
+            var reader = new Utf8JsonReader(json);
+            IJsonModel<T> instance = (T)Activator.CreateInstance(typeof(T), true)!;
```

Comment:
Are there any AOT concerns with using Activator here?

Reply:
Possibly, but this is what Michael is planning to use for serialization, and so if it's not AOT friendly, we have bigger problems than mocking. Moreover, I think mocking frameworks are fundamentally not AOT safe. Lastly, I don't think there are scenarios for AOT compiling tests.